### PR TITLE
Initial pass of Box/Folder models + behavior

### PR DESCRIPTION
**Pull request by tpendragon**
There's a lot to do here still, but it's a good starting point so we can
divide and conquer.

### CODE REVIEW

File: app/models/ephemera_folder.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#  `rails generate hyrax:work EphemeraFolder`
+class EphemeraFolder < ActiveFedora::Base
+  include ::Hyrax::WorkBehavior
+  include ::Hyrax::BasicMetadata
+  include ::StateBehavior
+  apply_schema EphemeraSchema, ActiveFedora::SchemaIndexingStrategy.new(
+    ActiveFedora::Indexers::GlobalIndexer.new([:symbol, :stored_searchable, :facetable])
+  )
+  # Change this to restrict which works can be added as a child.
+  # self.valid_child_concerns = []
```
Comment (escowles) [inline app/models/ephemera_folder.rb]: Do we want to limit to FileSet?

File: app/services/workflow/updated_event.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module Workflow
   class UpdatedEvent
     def self.call(target:, **)
+      return unless target.is_a?(ScannedResource) || target.is_a?(MultiVolumeWork)
```
Comment (escowles) [inline app/services/workflow/updated_event.rb]: Do we want to maintain the list of models here, or add something to the models that should generate events?

**Comment by escowles** [comment]
I created #1205 and #1206 for some related issues that we can resolve once this is merged.

File: app/models/ephemera_folder.rb (hunk 1 above)
Comment (tpendragon) [inline app/models/ephemera_folder.rb]: Nah. This was meant to populate the "Add Child Work" drop down.

**Comment by escowles** [comment]
I just pushed a commit that _should_ get us to 💯  — and then I think this is ready to merge.

**Pull request merged**